src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/AzureOpenAIEmbeddingProvider.java

```diff
@@ -115,6 +115,7 @@ public Uni<Response> vectorize(
       List<String> texts,
       Optional<String> apiKeyOverride,
       EmbeddingRequestType embeddingRequestType) {
+    checkEmbeddingApiKeyHeader(providerId, apiKeyOverride);
```

Comment:
Is override really always needed? (maybe it's just odd name for parameter and means "apiKey"?)